src/main/java/QueryIndex.java

```diff
@@ -74,14 +78,27 @@ public int compare(String o1, String o2) {
      * @throws ParseException when a query could not be parsed
      */
     public void queryMap(HashMap<Integer,String> queries,
-                         String indexDirectoryLocation) throws IOException, ParseException {
+                         String indexDirectoryLocation) throws Exception {
         Directory directory = FSDirectory.open(Paths.get(indexDirectoryLocation));
         DirectoryReader directoryReader = DirectoryReader.open(directory);
         IndexReader indexReader = DirectoryReader.open(directory);
         IndexSearcher indexSearcher = new IndexSearcher(directoryReader);
         indexSearcher.setSimilarity(AnalyzerSimilarityFactory.getSimilarity(mSimilarityString));
+
```

Comment:
I don't fully understand this code block, what's the distinction between the "500" passes in and the "70000" in the comparison? Do they represent different things?

Reply:
Actually, I am getting top 500 high freq words and then checking their freq is greater than 70000. After that check, 482 words remain. 500 is just something that needs to be passed to getHighFreqTerms function

Reply:
Makes sense, thanks! That seems like it should work to me so, not sure what the issue is unfortunately.